libs/core/kiln_ai/adapters/fine_tune/base_finetune.py

```
@@ -104,11 +110,32 @@ async def create_and_start(
             thinking_instructions=thinking_instructions,
             parent=parent_task,
             data_strategy=data_strategy,
+            run_config=run_config,
         )
 
         adapter = cls(datamodel)
         await adapter._start(dataset)
 
+        # Update the run config properties for fine-tuning
+        if run_config is not None:
+            run_config.model_provider_name = ModelProviderName.kiln_fine_tune
+            run_config.model_name = datamodel.model_id()
+            # Build the fine-tune prompt ID
+            task = datamodel.parent_task()
+            if task is None:
+                raise ValueError("Finetune must have a parent task")
+            project = task.parent_project()
+            if project is None:
+                raise ValueError("Task must have a parent project")
+            run_config.prompt_id = (
+                f"fine_tune_prompt::{project.id}::{task.id}::{datamodel.id}"
+            )
+            # follow the same behavior as provider_api where json_instructions is the default for fine-tuned models
+            run_config.structured_output_mode = (
+                datamodel.structured_output_mode
+                or StructuredOutputMode.json_instructions
```

P0: this isn't right. 

looks like we have this logic baked into the subclasses, so this is kinda implemented but with issues:
 - Example in Fireworks: `self.datamodel.structured_output_mode = StructuredOutputMode.json_mode`

Regardless, we shouldn't fallback here. It should be an error at creation time if we don't have a good structured_output_mode. We only know the right value now, and we'll never know it again. Should fail loudly not silently produce a model that doesn't work well.

P0s:
 - The provider specific subclasses should set the new field (run_config.structued_output_mode), not the old legacy field `datamodel.structured_output_mode`. Leave the legacy field empty going forward. 
 - We should error on missing value. Add an exception after _start the task is structured but the run_config.structued_output_mode field is blank.
 - Never fallback to json_instructions here! We need to save it now or exception.
 - don't set run_config.structured_output_mode unless the task is structured output (I don't see that check here)